libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -498,6 +498,11 @@ bool AP_Mission::start_command(const Mission_Command& cmd)
 ///     cmd.index is updated with it's new position in the mission
 bool AP_Mission::add_cmd(Mission_Command& cmd)
 {
+    // Add home if its not already present
+    if (_cmd_total < 1) {
```

Comment:
i think we need a check for index 0 in set_item, triggering add of home, return true, otherwise it could break currently working scripts

Reply:
Fixed